src/Transform.cpp

```diff
@@ -1,59 +1,61 @@
 #include "Transform.h"
 
+#include <glm/gtc/matrix_transform.hpp>
+
 namespace arch {
 
-std::pair<Quaternion, Quaternion> Transform::rotate(float angle, float x, float y, float z) {
+std::pair<Quaternion, Quaternion> Transform::rotate(f32 angle, f32 x, f32 y, f32 z) {
 	// normalize x,y,z
-	auto norm = static_cast<float>(sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2)));
+	auto norm = static_cast<f32>(sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2)));
 	x /= norm;
 	y /= norm;
 	z /= norm;
 
 	// fix angle
 	angle /= 2;
-	const double pi = std::acos(-1);
-	angle *= static_cast<float>(pi / 180.0);
+	const f64 pi = std::acos(-1);
```

Comment:
pi can be constexpr (gcc users will benefit from this), also you can use std::acosf and get rid of unnecessary conversion

Reply:
TBH i don't think, that this class will stay for long. Probably it'll be removed during Vulkan API implementation

Reply:
I think we should prevent this code from situation like // Temporary solution -> committed 5 years ago

Reply:
I that case i think the best solution it to remove a file.

Reply:
If you are sure it won't be necessary then please, do it